src/Intervals.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+/*
+ * This file is part of composer/semver.
+ *
+ * (c) Composer <https://github.com/composer>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Composer\Semver;
+
+use Composer\Semver\Constraint\Constraint;
+use Composer\Semver\Constraint\ConstraintInterface;
+use Composer\Semver\Constraint\EmptyConstraint;
+use Composer\Semver\Constraint\MultiConstraint;
+
+/**
+ * Helper class to evaluate constraint by compiling and reusing the code to evaluate
+ */
+class Intervals
+{
+    private static $intervalsCache = array();
+
+    /**
+     * @psalm-var array<string, int>
+     */
+    private static $opSortOrder = array(
+        '>=' => -2,
+        '<' => -1,
+        '==' => 0,
```

Comment:
Can drop `==` here as they get converted

Reply:
Why did you mark this resolved?

Reply:
Oh because I reused the array for another type of sorting at some point but I ended up dropping that. Will remove == and != too I guess can safely go.